backend/src/main/kotlin/code/nebula/cipherquest/repository/VectorStoreRepository.kt

```diff
@@ -11,4 +14,24 @@ class VectorStoreRepository(
         val sql = "SELECT EXISTS (SELECT 1 FROM vector_store WHERE metadata->>'file_name' = ?)"
         return jdbcTemplate.queryForObject(sql, Boolean::class.java, source)
     }
+
+    fun getMessageHistoryByUserId(userId: String?): List<Message> {
+        val sql =
+            "SELECT ROW_NUMBER() OVER (ORDER BY created_at) AS index, " +
+                "content as message, metadata->>'messageType' as sender, created_at as timestamp " +
+                "FROM vector_store " +
+                "WHERE metadata->>'conversationId' = ? ORDER BY created_at"
```

Comment:
We can leverage the multiline syntax.

```suggestion
        val sql =
            """
                SELECT ROW_NUMBER() OVER (ORDER BY created_at) AS index,
                content as message, metadata->>'messageType' as sender, created_at as timestamp
                FROM vector_store
                WHERE metadata->>'conversationId' = ? ORDER BY created_at
            """
```